lib/msf/core/db_manager/exploit_attempt.rb

```diff
@@ -27,186 +27,145 @@ def report_exploit_attempt(host, opts)
   }
   end
 
+  # Create an `Mdm::ExploitAttempt` (and possibly an `Mdm::VulnAttempt`, if
+  # the `vuln` option is passed).
+  #
+  # @option (see #do_report_failure_or_success)
+  # @return (see #do_report_failure_or_success)
   def report_exploit_failure(opts)
+    return unless opts.has_key?(:refs) && !opts[:refs].blank?
+    host   = opts[:host] || return
 
-  ::ActiveRecord::Base.connection_pool.with_connection {
-    wspace = opts.delete(:workspace) || workspace
-    mrefs  = opts.delete(:refs) || return
-    host   = opts.delete(:host)
-    port   = opts.delete(:port)
-    prot   = opts.delete(:proto)
-    svc    = opts.delete(:service)
-    vuln   = opts.delete(:vuln)
-
-    timestamp  = opts.delete(:timestamp)
-    freason    = opts.delete(:fail_reason)
-    fdetail    = opts.delete(:fail_detail)
-    username   = opts.delete(:username)
-    mname      = opts.delete(:module)
+    wspace = opts[:workspace] || workspace
+    port   = opts[:port]
+    prot   = opts[:proto] || "tcp"
```

Comment:
`"tcp"` should be moved into a constant so it's not just a magic string embedded in the method body.